### PR TITLE
RBSFXIRJTP 260 feature/sell product to market

### PR DESCRIPTION
**Pull request by pashunov151**
Implement a selling feature, which increases the specified products' quantities in both Market and Orderbook

### CODE REVIEW

**Review by Lilyanov — APPROVED** [review]

**Review by trayan-muchev — APPROVED** [review]

**Review by jelyslime — DISMISSED** [review]

**Review by trayan-muchev — DISMISSED** [review]

**Review by Lilyanov — DISMISSED** [review]

**Review by Lilyanov — APPROVED** [review]

File: calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java [hunk 1]
```diff
@@ -142,6 +146,60 @@ void buyProductShouldThrowProductNotAvailableException() {
...
+
+            //Act
+
+            transactionService.sellProduct(Collections.singletonList(itemDto));
+        });
+    }
+
+    @Test
+    void sellProductShouldThrowIllegalArgumentExceptionIfPriceIsLessOrEqualZero() {
+
```
Comment (gaantonova) [inline calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java:178]: delete empty line

File: calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java [hunk 2]
```diff
@@ -142,6 +146,60 @@ void buyProductShouldThrowProductNotAvailableException() {
...
+        //Act
+        String message = transactionService.sellProduct(Collections.singletonList(item));
+
+        //Assert
+        assertEquals(SUCCESSFUL_SELL_MESSAGE, message);
+    }
+
+    @Test
+    void sellProductShouldThrowIllegalArgumentExceptionIfQuantityIsLessOrEqualZero() {
+
```
Comment (gaantonova) [inline calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java:163]: delete empty line

File: calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java [hunk 3]
```diff
@@ -142,6 +146,60 @@ void buyProductShouldThrowProductNotAvailableException() {
...
+        //Assert
+        assertThrows(IllegalArgumentException.class, () -> {
+
+            //Arrange
+            ItemDto itemDto = new ItemDto(ITEM_NAME_CRUSTS, ITEM_PRICE, ITEM_ORIGIN, -ITEM_QUANTITY);
+
+            //Act
+
```
Comment (gaantonova) [inline calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java:171]: delete empty line

File: calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java [hunk 4]
```diff
@@ -142,6 +146,60 @@ void buyProductShouldThrowProductNotAvailableException() {
...
+        //Assert
+        assertEquals(SUCCESSFUL_SELL_MESSAGE, message);
+    }
+
+    @Test
+    void sellProductShouldThrowIllegalArgumentExceptionIfQuantityIsLessOrEqualZero() {
+
+        //Assert
+        assertThrows(IllegalArgumentException.class, () -> {
+
```
Comment (gaantonova) [inline calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java:166]: delete empty line

File: calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java [hunk 5]
```diff
@@ -142,6 +146,60 @@ void buyProductShouldThrowProductNotAvailableException() {
...
+            transactionService.sellProduct(Collections.singletonList(itemDto));
+        });
+    }
+
+    @Test
+    void sellProductShouldThrowIllegalArgumentExceptionIfPriceIsLessOrEqualZero() {
+
+        //Assert
+        assertThrows(IllegalArgumentException.class, () -> {
+
```
Comment (gaantonova) [inline calculator/src/test/java/com/market/banica/calculator/service/TransactionServiceImplTest.java:181]: delete empty line

**Pull request merged**